cache/cache.go

```diff
@@ -96,8 +97,9 @@ func (c *lruCache) Len() int {
 }
 
 func (c *lruCache) Remove(key []byte) Node {
-	if elem, exists := c.dict[string(key)]; exists {
-		return c.remove(elem)
+	keyS := string(key)
+	if elem, exists := c.dict[keyS]; exists {
```

Comment:
This undoes the zero byteslice->string Go map lookup optimization.

Reply:
Oh wow, didn't know about this compiler optimization.